pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingManager.java

```diff
@@ -616,4 +620,14 @@ void refreshSegment(String segment) {
       }
     }
   }
+
+  @Override
+  public Map<String, ServerInstance> getEnabledServerInstanceMap() {
+    return new HashMap<>(_enabledServerInstanceMap);
```

Comment:
Why copy ? Caller will modify ?

Reply:
at this moment no but there's a chance the enabledServerInstanceMap changed during cluster change handler that modifies the routing table.

Reply:
this is resolved. we decided to let the caller replicate the map